spec/ysh-source.test.sh

```diff
@@ -0,0 +1,24 @@
+# spec/ysh-source
+
+## our_shell: ysh
+## oils_failures_allowed: 0
+
+#### --builtin flag
+source --builtin stdlib/math.ysh
```

Comment:
I wonder if we should make `stdlib/` the implicit root?   So it's just `source --builtin math.ysh`

I think by definition, `stdlib/` is exactly the things that are built into the binary